### PR TITLE
[CMR] M3-4428: MUI Mobile Nav

### PR DESCRIPTION
**Pull request by tiffwong**
## Description
Replacing the Reach UI mobile nav with MUI components to improve keyboard navigation

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers
Please check with different browsers and voiceover software

### CODE REVIEW

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 1]
```diff
@@ -154,107 +105,167 @@ const useStyles = makeStyles((theme: Theme) => ({
...
-            </MenuPopover>
-            <Backdrop className={classes.settingsBackdrop} open={isExpanded} />
-          </>
-        );
-      }}
-    </ReachMenu>
+          // Otherwise return a NavGroup (dropdown menu)
+          return (
+            <>
+              <ListItem
```
Comment (WilkinsKa1) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:236]: Let's add:
aria-haspopup="true"
aria-controls={menu id here}, could use something like: {`menu-${thisGroup.group`}
id={button id here}, similar idea here, {`button-${thisGroup.group`}

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 2]
```diff
@@ -154,107 +105,167 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+              >
+                <ListItemText primary={thisGroup.group} />
+                {isGroupOpen(thisGroup.group) ? (
+                  <ExpandLess className={classes.caret} />
+                ) : (
+                  <ExpandMore className={classes.caret} />
+                )}
+              </ListItem>
+              <Collapse in={groupMap[thisGroup.group]}>
+                <List component="div" disablePadding>
```
Comment (WilkinsKa1) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:250]: role="menu" 
id={menu id here} 
aria-labelledby={menu button id here}>

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 3]
```diff
@@ -154,107 +105,167 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+                  {filteredLinks.map((thisLink: any) => (
+                    <ListItem
```
Comment (WilkinsKa1) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:252]: role="menuitem"

**Review by WilkinsKa1 — COMMENTED** [review]
Excellent work on this, nav is working great 👏  I added some non-blocking feedback for some additional aria attributes that we were gaining from Reach. We may also want to consider a separate description to give users instructions on how to access/traverse the list (arrow key/tab to navigate, enter to activate, etc.)

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 4]
```diff
@@ -154,107 +105,177 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+  const [openMonitors, setOpenMonitors] = React.useState(false);
+
+  const groupMap = {
+    Compute: openCompute,
+    Network: openNetwork,
+    Storage: openStorage,
+    Monitors: openMonitors
+  };
+
+  const toggleMenu = () => {
```
Comment (Jskobos) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:136]: little verbose, could do `setOpen(!open)`, or to be safer `setOpen(isOpen => !isOpen)`

**Review by Jskobos — COMMENTED** [review]
Voiceover/functionality all working great.

Questions!:

1. I can't close the menu by escaping or clicking/touching outside the boundary. Is this planned/possible?
2. State is preserved after closing the menu, so if I close and re-open the menu the submenu I was looking at is still open. Was this intentional? I don't mind it, just curious.

**Review by DevDW — COMMENTED** [review]
This is working well overall, but for some reason in Safari, I couldn't use keyboard navigation to get to the mobile nav. When I tabbed, it jumped from the address bar straight to Pending Actions, and vice versa when I reverse tabbed.

**Review by Jskobos — APPROVED** [review]

**Review by WilkinsKa1 — APPROVED** [review]
I've tested this in chrome emulator, with JAWS enabled, and on an actual device with a11y settings on. This is working really well, great work on getting this sorted!! 🎉 

Only additional feedback I have after testing on a device is to increase the padding of the nested items (they're a little tricky to tap, and with a11y settings require a double tap to activate)- approving in advance of that!

**Comment by tiffwong** [comment]
> This is working well overall, but for some reason in Safari, I couldn't use keyboard navigation to get to the mobile nav. When I tabbed, it jumped from the address bar straight to Pending Actions, and vice versa when I reverse tabbed.

Turns out this is a setting you change to change in Safari lol
To enable it, check "Preferences > Advanced > Press tab to highlight each item on a page"

**Pull request merged**